src/toolkits/style_transfer/style_transfer.cpp

```diff
@@ -686,6 +686,12 @@ void style_transfer::init_train(gl_sarray style, gl_sarray content,
   }
   size_t num_styles = num_styles_iter->second;
 
+  auto pretrained_weights_iter = opts.find("pretrained_weights");
```

Comment:
Is `auto` helping or hurting readability here?

Reply:
I believe the alternative is
```
std::map<std::string, flexible_type>::iterator pretrained_weights_iter = opts.find("pretrained_weights");
```
So I would argue yes.